test/client/fixtures/sandbox/fetch-test.js

```diff
@@ -332,23 +333,51 @@ if (window.fetch) {
     });
 
     module('regression', function () {
-        test('request promise should be rejected for the 500 http status code (GH-602)', function () {
-            return fetch('/respond-500')
-                .then(function (response) {
-                    return response.text();
-                })
-                .then(function () {
-                    ok(false, 'request promise should be rejected');
-                })
-                .catch(function (err) {
-                    ok(err instanceof TypeError);
-                });
+        test('request promise should emulate native behavior with error and error status code (GH-1397)', function () {
```

Comment:
`should emulate a native browser behavior for fetch request with non success status code`